mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -298,11 +298,7 @@ class LinearRegressionModel private[ml] (
    */
   // TODO: decide on a good name before exposing to public API
   private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
-    val t = udf { features: Vector => predict(features) }
-    val predictionAndObservations = dataset
-      .select(col($(labelCol)), t(col($(featuresCol))).as($(predictionCol)))
-
-    new LinearRegressionSummary(predictionAndObservations, $(predictionCol), $(labelCol))
+    new LinearRegressionSummary(transform(dataset), $(predictionCol), $(labelCol))
```

Comment:
Are there other cases where a missing predictionCol will cause errors?  We should at least do the "easy option" above.